File: src/modifiers/task.ts
import { createModifier } from './createModifier';

/// SwiftUI `.task(priority:_:)`. Fires `handler` once when the view appears
/// (the bridged async task simply invokes the JS callback). Use it for
/// load-on-appear side effects.
///
/// ```tsx
/// <Text modifiers={[task(() => load())]} />
/// ```
export function task(handler: () => void) {
  return createModifier('task', { handler });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/Task.swift
import SwiftUI

/// SwiftUI `.task(_:)`. Runs once when the view appears; the bridged async
/// task simply invokes the JS `handler` (it does no awaiting itself).
enum RNWTaskModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("task") { view, params, ctx in
            let handlerId = params.int("handler")
            return AnyView(view.task {
                ctx.fire(handlerId)
            })
        }
    }
}
